named_entity_recognition_bert/Legal-NER.ipynb

```diff
@@ -0,0 +1,1802 @@
+     ]
+    }
+   ],
+   "source": [
+    "label2id = {k: v for v, k in enumerate(sorted(unique_labels))}\n",
+    "id2label = {v: k for v, k in enumerate(sorted(unique_labels))}\n",
+    "print(\"id2label: \", id2label)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8149d73e",
+   "metadata": {},
+   "source": [
+    "### 3. Prepare data for Bert NER model\n",
+    "<a id='prepare_data_for_bert_model'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e7125cf-7d46-4ff6-b111-35112c645146",
+   "metadata": {},
+   "source": [
+    "We will use pretrained tokenizer bert-base-cased for our data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "deebb946",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tokenizer = BertTokenizerFast.from_pretrained('bert-base-cased')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "082d6c6d-fa38-4686-a3a0-476d71d93c92",
+   "metadata": {},
+   "source": [
+    "We should align named entities that we downloaded in [SuperAnnotate format](https://doc.superannotate.com/docs/sdk-export-annotations) with tokens we get from tokenizer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "c1dad3c3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def align_label(txt_tokenized,\n",
+    "                entities,\n",
+    "                label2id):\n",
+    "    label_ids = []\n",
+    "    cnt = 0\n",
+    "    for word_idx, (start,end) in zip(txt_tokenized.word_ids(), txt_tokenized['offset_mapping'][0]):\n",
+    "        if word_idx is None:\n",
+    "            label_ids.append(-100)\n",
+    "            continue\n",
+    "        found_entity = False\n",
+    "        for entity in entities:\n",
+    "            if entity['start'] <= int(start) and entity['end'] >= int(end) and not found_entity:\n",
+    "                label = entity['className']\n",
+    "                label_ids.append(label2id[label])\n",
+    "                found_entity = True\n",
+    "                break\n",
+    "        if not found_entity:\n",
+    "            label_ids.append(label2id['O'])\n",
+    "\n",
+    "    return label_ids"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "142f392e-e19a-45c5-a458-d3f9003b78ee",
+   "metadata": {},
+   "source": [
+    "And now we can create class for our dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "789fc352",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class DataSequence(torch.utils.data.Dataset):\n",
+    "\n",
+    "    def __init__(self, texts, entities=None, label2id=None):\n",
+    "        if not entities:\n",
+    "            entities = [[] for t in texts]\n",
+    "        self.texts = [tokenizer(str(i),\n",
```

Comment:
Let's do in other way

configured_tokenizer = lambda text: tokenzier(str(text),  padding='max_length', ... )
self.texts = [configured_tokenizer(text) for text in texts]

Reply:
+